wagtail/snippets/views/snippets.py

```diff
@@ -1108,6 +1117,40 @@ def chooser_viewset(self):
             per_page=self.chooser_per_page,
         )
 
+    def get_list_display(self, request):
+        """
+        Return a sequence containing the fields/method output to be displayed
+        in the list view.
+
+        For more details, see
+        :attr:`~wagtail.snippets.views.snippets.SnippetViewSet.list_display`.
+        """
+        if not self.list_display and self.index_view_class.list_display:
+            self.list_display = self.index_view_class.list_display.copy()
+            if self.draftstate_enabled:
+                self.list_display += [LiveStatusTagColumn()]
+        return self.list_display
+
+    def get_list_filter(self, request):
+        """
+        Returns a sequence containing the fields to be displayed as filters in
+        the right sidebar in the list view.
+
+        For more details, see
+        :attr:`~wagtail.snippets.views.snippets.SnippetViewSet.list_filter`.
+        """
+        return self.list_filter
+
+    def get_filterset_class(self):
```

Comment:
Decided not to add `request` here because we're already passing the request to the FilterSet when we instantiate it in the generic IndexView. This means request will be available in the FilterSet instance, so I don't see why people would need to customise the class based on the request.

Also, FilterSet isn't a thing in ModelAdmin, so we don't have to follow its existing API (unlike `get_list_display()` and `get_list_filter()` that still take request as parameter for that reason).